setup.py

```diff
@@ -230,81 +222,25 @@ def build_cmake_extension(self, ext: CMakeExtension):
-        # Run the original build_ext command
-        super().run()
-
-        # Construct library name based on ext suffix (contains python version, architecture and .so)
-        library_name = "libcustom_calls.so"
-
-        package_root = os.path.dirname(__file__)
-        frontend_path = glob.glob(
-            os.path.join(package_root, "frontend", "**", library_name), recursive=True
-        )
-        build_path = glob.glob(os.path.join("build", "**", library_name), recursive=True)
-        lib_with_r_path = "@rpath/libcustom_calls.so"
-
-        original_path = frontend_path[0] if frontend_path else build_path[0]
-
-        # Run install_name_tool to modify LC_ID_DYLIB(other the rpath stays in vars/folder)
-        subprocess.run(
-            ["/usr/bin/install_name_tool", "-id", lib_with_r_path, original_path],
-            check=False,
-        )
-
-
 # Compile the library of custom calls in the frontend
 if system_platform == "Linux":
-    custom_calls_extension = Extension(
-        "catalyst.utils.libcustom_calls",
-        sources=[
-            "frontend/catalyst/utils/libcustom_calls.cpp",
-            "frontend/catalyst/utils/jax_cpu_lapack_kernels/lapack_kernels.cpp",
-            "frontend/catalyst/utils/jax_cpu_lapack_kernels/lapack_kernels_using_lapack.cpp",
-        ],
-        extra_compile_args=["-std=c++17"],
-    )
-    cmdclass = {"build_ext": CustomBuildExtLinux}
+    cmdclass = {"build_ext": CMakeExtension}
 
 elif system_platform == "Darwin":
-    variables = sysconfig.get_config_vars()
-    # Here we need to switch the deault to MacOs dynamic lib
-    variables["LDSHARED"] = variables["LDSHARED"].replace("-bundle", "-dynamiclib")
-    if sysconfig.get_config_var("LDCXXSHARED"):
-        variables["LDCXXSHARED"] = variables["LDCXXSHARED"].replace("-bundle", "-dynamiclib")
-    custom_calls_extension = Extension(
-        "catalyst.utils.libcustom_calls",
-        sources=[
-            "frontend/catalyst/utils/libcustom_calls.cpp",
-            "frontend/catalyst/utils/jax_cpu_lapack_kernels/lapack_kernels.cpp",
-            "frontend/catalyst/utils/jax_cpu_lapack_kernels/lapack_kernels_using_lapack.cpp",
-        ],
-        extra_compile_args=["-std=c++17"],
-    )
-    cmdclass = {"build_ext": CustomBuildExtMacos}
+    cmdclass = {"build_ext": CMakeExtension}
 
+    # variables = sysconfig.get_config_vars()
+    # Here we need to switch the deault to MacOs dynamic lib
+    # variables["LDSHARED"] = variables["LDSHARED"].replace("-bundle", "-dynamiclib")
+    # if sysconfig.get_config_var("LDCXXSHARED"):
+    #    variables["LDCXXSHARED"] = variables["LDCXXSHARED"].replace("-bundle", "-dynamiclib")
+    # cmdclass = {"build_ext": CustomBuildExtMacos}
 
 project_root_dir = os.path.abspath(os.path.dirname(__file__))
 frontend_dir = os.path.join(project_root_dir, "frontend")
 
 ext_modules = [
-    custom_calls_extension,
+    CMakeExtension("catalyst.utils.libcustom_calls", sourcedir=frontend_dir),
```

Comment:
TODO: Indentify if this is intended to be a module (so) or a dynamic library (dylib)